transcom/mymove
# Main b 19788 backend - office user ability to manage docs

## [B-19788](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19788)

INT PRs:
https://github.com/transcom/mymove/pull/13140
https://github.com/transcom/mymove/pull/13164

## Summary

This is backend work to give office users the ability add/delete documents(orders, amended orders, and additional documents). The api for adding and delete documents have been copied over from the `internal api` to `ghc api`. `deletedOn` and `uploadType` has been added to the upload object which will indicate if the file have been deleted and who uploaded the file(USER, OFFICE, PRIME). The Delete functionally is a soft delete - meaning it will keep the file, but have `deletedOn` date in the database.

### How to test

NOTE: I've tried using Swagger UI to upload files and was getting Unauthorized error so I had to use Postman for the test.

1. As a customer -> create and finish submitting a move.
4. Using Postman -> Import new `ghc.yaml` from swagger folder -> Using api, upload/delete files and refresh the page to see changes.(Check below for Postman apis.)
5. As a customer -> Check the related documents to see that the changes have been made. (Check below for instructions on how to check)

#### To use in Postman:
You can find premade settings for these in the Postman Collections after you import.
For OrderID -> get the id from dev tools -> order GET request
For UploadID -> get the id from dev tools -> id in the `Uploads` object(check the screenshots below).
For DocumentID -> get the id from dev tools -> id under `uploaded_orders` object(see screenshots below)

When uploading new file -> Go to "Body" Tab in Postman under URL and set file to "File" instead of "Text" under Key column. Then click on the "Select files" under Value column to choose your file.

Upload order:
`http://officelocal:3000/swagger-ui/ghc.html#/uploads/createUpload`
Upload amended order:
`http://officelocal:3000/swagger-ui/ghc.html#/order/uploadAmendedOrders`
Upload additional documents:
`http://officelocal:3000/swagger-ui/ghc.html#/move/uploadAdditionalDocuments`
Delete any files:
`http://officelocal:3000/swagger-ui/ghc.html#/uploads/deleteUpload`


Places to check as a Customer:
Orders: 
1. Go to move.
2. Click "Review your request" button.
3. Click "Edit" next to Orders and you will see uploads at the bottom.

Amended Orders:
1. Go to move.
2. Click "Upload documents" next to Orders.

Additional Documents:
1. Go to move.
2. Click "Upload Additional Documents" button.

## Screenshots
Internal api - Orders GET
![image](https://github.com/transcom/mymove/assets/146856854/9f10c724-228f-48c4-a83f-49e2450158d3)

Internal api - Moves GET
![image](https://github.com/transcom/mymove/assets/146856854/725c1557-f00c-42bb-9c74-759f5023e748)
